### DISCUSSION TITLE
Session Chat Log

### DISCUSSION THREAD

**Discussion by gene96817** [jamulussoftware/jamulus#1255] [Ideas]
Proposal:
Maintain a session chat log on the server. This would be kept when the active channels is non-zero.
The log would flush when the active channels goes to zero,

Rational:
When a session starts (one channel  active), as more channels become active, any chat is only visible to the connected channel. A person joining late cannot see the earlier chat. This is disruptive if there are instructions or information distributed by chat. This log would allow a late participant to catch up on earlier chat.

**Comment by corrados** [comment]
This is a duplicate, see https://github.com/corrados/jamulus/issues/276

**Comment by corrados** [comment]
Here is the corresponding pull request: https://github.com/corrados/jamulus/pull/844

**Comment by ann0see** [comment]
> A person joining late cannot see the earlier chat. This is disruptive if there are instructions or information distributed by chat. 

I get your point. Concerning larger groups. 

But many chat programs also don't allow someone to read chat messages from before they joined. I personally don't really like the idea that someone is able to read chat messages from before he/she joined for privacy reasons. 

Maybe a one session welcome message would be good. This allows "private" chat messages and also announcements

**Comment by corrados** [comment]
See my comment here: https://github.com/corrados/jamulus/issues/276#issuecomment-633220244

**Comment by pljones** [comment]
Another point against having this on the server is that is someone wants server-based chat, they can use something like Discord.

**Comment by gene96817** [comment]
I agree we can use a different program for a chat channel. However, we are already working hard to introduce Jamulus to less technical users. Juggling multiple applications and windows pushes their ability to manage the complexity. 

Privacy makes sense for personal chatter. For session related information and coordination, there should be no privacy issues. Maybe we support this function for session management. Personal chatter and gossip needs privacy and it should be on Discord or another private channel.

This feature doesn't need storage (persistence) after the session. There are no long term storage issues.

**Comment by ann0see** [comment]
> This feature doesn't need storage (persistence) after the session.

Would it be possible to make the welcome message or similar for announcements which are only shown in this session?

**Comment by gene96817** [comment]
Using the welcome message for session announcements requires one of the session participant to be a server admin. It would work for users that "borrow" our server. This kind of feature requires every group to have their own server. I though we were concerned about having lots of underutilized servers.

**Comment by pljones** [comment]
> Privacy makes sense for personal chatter. For session related information and coordination, there should be no privacy issues.

How does the software determine the difference?  Entire teams of legal experts generally take years agreeing on things like this...

**Comment by gene96817** [comment]
private chatter is not responsibility of Jamulus. Why isn't problem solved by saying the chat window is public and no private information should be put into chat. No one can prevent a person to disclose private information. Privacy protection applies to data that is expected to be private.

**Comment by corrados** [comment]
> This feature doesn't need storage (persistence) after the session.

How would you define an end of a session? If you have a server which has a client connected as a streamer, this client will stay connected all the time. In that case your session would last forever.

> For session related information and coordination, there should be no privacy issues.

I disagree. I do not want to store any chat messages on my server. E.g., someone might post a URL to an illegal web site and disconnects. If another user connects to that server, you (the server operator) now provide him this illegal link. Also, if personal information is posted (tel. number, etc.) it is then also stored on your server.

Storing chat messags on the server has more problems than it solves. For me it is a no-go to store any chat message on a Jamulus server.

**Comment by gilgongo** [comment]
This doesn't seem be an issue that will arrive at a resolution any time soon. Moving to discussion.
